logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineBus.java

```diff
@@ -81,11 +81,10 @@ public void sendEvents(final PipelineOutput sender,
                         if (lastResponse != null && lastResponse.getStatus() == PipelineInput.ReceiveStatus.FAIL) {
                             // when last call to internalReceive generated a fail, restart from the
                             // fail position to avoid reprocessing of some events in the downstream.
-                            lastFailedPosition = lastResponse.getSequencePosition();
-
-                            logger.warn("Attempted to send event to '{}' but that address reached error condition. " +
-                                    "Will Retry. Root cause {}", address, lastResponse.getCauseMessage());
-
+                            // we get 0~clones.length position on failure, keep tracking failed position to correctly copy from original orderedEvents
```

Comment:
Should this read `clones.length`?

Reply:
I think this is meant to express the range of `0` to `clones.length` (differentiated from the range `0` to `orderedEvents.length`), but I have addressed it in e2cf5a03e